src/App.tsx

```diff
@@ -1,61 +1,131 @@
+import { useState } from 'react';
 import './App.scss';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [newTodos, setNewTodos] = useState(todos);
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorSelect, setErrorSelect] = useState(false);
+
+  const allTodosIds = [...todos].map(el => el.id);
+  const newId = Math.max(...allTodosIds) + 1;
+  const newUser = getUserById(selectedUserId);
+
+  const createNewTodo = () => {
+    return {
+      id: newId,
+      title,
+      completed: false,
+      userId: selectedUserId,
+      user: newUser,
+    };
+  };
+
+  const newTodo = createNewTodo();
+
+  const addNewTodo = () => {
+    return setNewTodos([
+      ...newTodos,
+      newTodo,
+    ]);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setErrorTitle(true);
+    }
+
+    if (!selectedUserId) {
+      setErrorSelect(true);
+    }
+
+    if (title.length > 0
+      && selectedUserId !== 0) {
```

Comment:
```suggestion
      && selectedUserId !== defaultUserId) {
```
`0` can be moved to the variable